### PR TITLE
sparse cat cross entropy

### PR DESCRIPTION
**Pull request by g1y5x3**
Updated `docs/quickstart.md`, `examples/serious_mnist.py`, `extra/training.py` accordingly after adding the function into `tensor.py`
~~Also, I deleted `examples/hlb_cifar10_torch.py` since its not going to be further updated.~~

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/runtime/ops_torch.py              1               1
tinygrad/tensor.py                         6               0
------------------------------------------------------------
total                                      7               1
------------------------------------------------------------
lines added in the tinygrad folder: 6
```

File: tinygrad/runtime/ops_torch.py [hunk 1]
```diff
@@ -26,5 +26,5 @@ def __init__(self, size:int, dtype:DType, buf:Optional[torch.Tensor]=None): supe
   def fromCPU(cls, x):
     buf = torch.from_numpy(x).requires_grad_(False).to(device)
     return cls(prod(x.shape), type_map[buf.dtype], buf)
-  def toCPU(self): return self._buf.cpu().numpy()
+  def toCPU(self): return self._buf.numpy()
```
Comment (dc-dc-dc) [inline tinygrad/runtime/ops_torch.py:29]: I think cpu is still needed here in case the torch tensor is not on the cpu

Comment (g1y5x3) [reply tinygrad/runtime/ops_torch.py:29]: Thanks for the catch. Will it make sense to just use `force=True`? Which would do
`t.detach().cpu().resolve_conj().resolve_neg().numpy()`

Comment (dc-dc-dc) [reply tinygrad/runtime/ops_torch.py:29]: could just leave it as is, don't think any issues has been brought up that requires the force flag addition here

**Comment by dc-dc-dc** [comment]
can you add some tests for `sparse_categorical_crossentropy` in test_ops, torch has there implementation under `torch.nn.functional.cross_entropy`

File: tinygrad/runtime/ops_torch.py (hunk 1 above)
Comment (geohot) [reply tinygrad/runtime/ops_torch.py:29]: This is the torch buffer, not the tinygrad buffer!

**Comment by geohot** [comment]
This should be two PRs I think

**Comment by g1y5x3** [comment]
split the `.cpu().numpy() -> .numpy()` part into #1594

**Comment by g1y5x3** [comment]
the current implementation matches the default setting in pytorch `reduction='mean'`

**Comment by geohot** [comment]
This is broken. Accuracy stays pegged at 1.00

`kafka@tubby:~/fun/tinygrad$ python3 test/models/test_mnist.py TestMNIST.test_conv
loss 0.39 accuracy 1.00:  52%|██████████████████████████████████████████████████████████████████████████████████████████████████▊                                                                                           | 52/100 [00:01<00:01, 37.02it/s]`
